week4/Q1.py

```diff
@@ -0,0 +1,104 @@
+def _rename_islands(islands_map, old_name, new_name):
+	'''
+	Changes names of the tiles.
+
+	Args:
+		islands_map: two-dimensional list
+		old_name: int
+		new_name: int
+
+	Returns:
+		two-dimensional list
+	'''
+	for i, row in enumerate(islands_map):
+		for j, tile in enumerate(row):
+			if tile == old_name:
+				islands_map[i][j] = new_name
+	return islands_map
+
+
+def _name_island(islands_map, row, column, islands, island_counter, upper_tile, left_tile):
+	'''
+	Gives name (which island) to the new tile, renames old tiles if needed (if islands which were seen as separate actually happen to be connected).
+
+	Args:
+		islands_map: two_dimensional list
+		row: int, row of the tile
+		column: int, column of the tile
+		islands: set, already existing islands
+		island_counter: int, last name given
+		upper_tile: int, value of the tile under which is new one
+		left_tile: int, value of the tile right to which is new one
+
+	Returns:
+		two-dimensional list, set, int
+	'''
+	if upper_tile:
+		islands_map[row][column] = upper_tile
+
+	if left_tile:
+		if islands_map[row][column] == 0:
+			islands_map[row][column] = left_tile
+		elif islands_map[row][column] != left_tile:
+			islands.discard(left_tile)
+			islands_map = _rename_islands(islands_map, left_tile, islands_map[row][column])
+
+	if islands_map[row][column] == 0:
+		island_counter += 1
+		islands_map[row][column] = island_counter
+		islands.add(island_counter)
+
+	return islands_map, islands, island_counter
+
+
+def count_islands(tiles_map, n_rows, n_columns):
+	'''
+	Counts number of islands on the given map.
+
+	Args:
+		tiles_map: two-dimensional list of bools
+		n_rows: int
```

Comment:
While the assignment explicitly mentioned having number of rows and columns as arguments, it is not strictly necessary in your implementation. Because the size of the map is given by `tiles_map`.

This is an example of how the requirements have changed from the original problem formulation to when you actually write the solution (there are solutions that do require the number of rows and columns, but in your case, the code does not require them).